src/main/java/com/google/sps/servlets/DashboardServlet.java

```diff
@@ -0,0 +1,69 @@
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.sps.servlets.Property;
+import java.util.HashMap;
+
+@WebServlet("/dashboard")
+public class DashboardServlet extends HttpServlet {
+
+ /**
+  * Given the correlator, this will create an entry in the database with default
+  * values.
+  *
+  * @param request This includes a correlator for the current user.
+  * @param response A 200 status or 400 error.
+  */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query interactionQuery = new Query(DBUtilities.INTERACTION_TABLE);
+    
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery interactions = datastore.prepare(interactionQuery);
+
+    int findNearestLocationCounter = 0;
+    int grantsLocationCounter = 0;
+    int interactsWithMapCounter = 0;
+    int skipToContentCounter = 0;
+    int returnToAdCounter = 0;
+    
+    for (Entity entity : interactions.asIterable()) {
```

Comment:
You can consider looking into stream filter methods to simplify some of this

https://beginnersbook.com/2017/10/java-8-stream-filter/

Reply:
I used an alternative method here.